### ISSUE TITLE
suppress console.log error messages during vat replay

### ISSUE THREAD

**Issue by warner** [Agoric/Agoric#8572]
## What is the Problem Being Solved?

The code inside vats sometimes emit error messages, printing things to `console.log()`. Although not part of the (consensus) transcript, these console messages are delivered to the kernel through the same pathway as syscalls (through the netstring pipe). When an offline vat is paged-in (including at kernel reboot time, when we must page in lots of vats), and we re-execute the vat's recent transcript entries, these `console.log()` calls happen again, causing the same console messages to be sent to the kernel.

For the benefit of node operators (and especially folks running full-kernel unit tests), the kernel prints these console messages to the kernel's stdout (specifically to the host application's stdout). This is a little confusing: it's not the *kernel* that is experiencing an error, but rather some vat that the kernel is supporting. But the alternative is to stash the error message in a file somewhere (like the database of errors that I'm interested in building). We decided that printing the error text to stdout is a good tradeoff between surfacing problems and causing confusion.

However, when these console messages are the result of transcript replay, the tradeoffs are different. Rebooting a kernel causes error messages to appear from days or weeks earlier (or even months, in the case of seldom-used vats, like vat-bootstrap). The error messages don't have timestamps on them, and are not correlated with anything happening on the chain right now. So these error messages are super-confusing.

## Description of the Design

I think the best fix is to change the kernel to stop printing vat `console` messages when they occur during a replay.

We already use a specialized syscall handler during replay to service the requests from the transcript, rather than executing the syscalls for real. If `console` messages were treated exactly like syscalls, this would be a good place to ignore those messages instead of printing them.

However, I think the console messages are split off earlier, inside the worker-manager code, using some form of console/logger object (maybe called `vatConsole`? maybe using the `anyconsole` package?).

So we might either want to add a method to the manager that lets us switch console-message-printing on and off (and start with it off, and turn it on only after we're done with transcript replay), or move the what-to-do-with-console-messages code onto the syscall handler (next to `doSyscall`, rather than inside it, so we aren't trying to put console messages in transcripts). The latter is probably cleaner, but might be more refactoring work.

## Security Considerations

none

## Scaling Considerations

none

## Test Plan

It might be nice to have a unit test that replaces the `console.log` used by the kernel-side code, and confirm that replay does not call it, but it might also be sufficient to perform one-time manual testing.

## Upgrade Considerations

None, the console messages are not part of consensus, and are not included in a transcript or stored anywhere.

## Debugging Considerations

I think it remains important to display vat error messages on the host application console during their original execution. This isn't mandatory, but we're currently limited to a single stdout/stderr stream, and we've all been frustrated by an execution platform which "swallows the error" in the code it's hosting, wasting hours while we try to figure out why progress isn't being made, and where seeing the error would have solved the problem in an instant. So for the benefit of people running unit tests that execute buggy vat code, I think we need to keep displaying them.

But, it'd be fun to think about some sort of fancy GUI execution shell, where the kernel's own output is in one window, and each vat's output (and errors) are in separate windows, so it'd be obvious which is which. Another form of this might be the "error database" mentioned above, in which stdout would see a single line like "XYZError in vat14-zoe, run 'sserror GUID' for more info", and the `sserror` command would query the DB for everything related to that error. This might be a better way to capture the cross-vat error propagation (and deep stacks), than attempting to print everything to a single stdout stream (and hoping humans can figure out how they're correlated).

**Comment by mhofman** [comment]
I can confirm that the new logs don't include any output from replaying vats during kernel start:
```
Sun, 18 Feb 2024 05:14:18 GMT
#21 3.455 2024-02-18T05:14:18.469Z launch-chain: Launching SwingSet kernel
Sun, 18 Feb 2024 05:14:51 GMT
#21 36.36 2024-02-18T05:14:51.377Z block-manager: block 967 begin
Sun, 18 Feb 2024 05:14:51 GMT
#21 36.42 2024-02-18T05:14:51.435Z SwingSet: vat: v1: evaluateBundleCap { manifestBundleRef: { bundleID: 'b1-773d4b972ad4fe803de4947c7f907a85a52fc08e1f6789ba1b853db4e691d10c43847612de281b4567710c92a678cf5ee9fa441d71f81a1f2bc538d6f255c75b' }, manifestGetterName: 'getManifestForUpgradeWallet', vatAdminSvc: Promise [Promise] {} }
```

I am however wondering if we should add a log entry after cosmic-swingset has finished launching the kernel as right now if things get stuck, it'd be hard to determine if we got stuck in kernel or cosmos land.

**Comment by mhofman** [comment]
Simple approach in #8943 closes this
